openmls/src/group/tests_and_kats/tests/capabilities_check.rs

```diff
@@ -0,0 +1,352 @@
+use crate::prelude::*;
+use crate::test_utils::single_group_test_framework::*;
+use crate::treesync::errors::LeafNodeValidationError;
+
+// Helper macro for checking error matches a provided pattern
+#[cfg(test)]
+macro_rules! assert_err_matches {
+    ($err:expr, $pattern:pat) => {
+        assert!(matches!($err.expect_err("Expected an error"), $pattern));
+    };
+}
+
+// Function to check that the correct error type was returned
+#[cfg(test)]
+fn test_valn0104_error<Provider: OpenMlsProvider>(error: Result<(), GroupError<Provider>>) {
```

Comment:
This function name sounds a bit like this is a test. Maybe call it `expect_valn0104_error`?